### PR TITLE
Add TypeScript configuration layer

### PR DESCRIPTION
**Pull request by cbowdon**
This adds support for TypeScript editing via [typescript-tools](https://github.com/clausreinke/typescript-tools) and [emacs-tss](https://github.com/aki2o/emacs-tss). Let me know if I've got anything wrong or could improve anything, I'll be happy to fix it.

### CODE REVIEW

**Comment by BrianHicks** [comment]
Some keybinding changes (see the [conventions](https://github.com/syl20bnr/spacemacs/blob/master/doc/CONVENTIONS.md), which I just found out about myself!)

 - `m h` should probably be `m h h`
 - `m d` should be `m g g` (think **g**oto, instead of **d**ebug)
 - `m c f` should be `m c b` (**c**ompile **b**uffer)

Is there a mismatch with the current npm version vs from git? It might be better, long-term, to instruct users to download the released version to avoid getting bugs.

Thanks for the layer!

File: contrib/lang/typescript/packages.el [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+(defun typescript/init-tss ()
+  "Initialize my package"
+  (use-package tss
+    :mode ("\\.ts\\'" . typescript-mode)
+    :defer t
+    :init
+    (progn
+      (require 'typescript)
+      (require 'tss)
```
Comment (BrianHicks) [inline contrib/lang/typescript/packages.el]: these shouldn't need to be required here, that should be set up by `user-package`. If you're running into trouble with undefined names, put that code under `:config`.

**Comment by cbowdon** [comment]
Thanks for the feedback, I'll make those changes and force push them up.

There _is_ a mismatch between the current typescript-tools version on NPM and the git repo, as the NPM version does not support the current TypeScript version. Hopefully there will be a new release of typescript-tools in the near(ish) future, at which point I agree the instructions should be updated.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
I tweaked the PR a bit in 37e515a57c6279cd519a9caa3a47b6856c751877 see the commit message for more info.
Cherry-picked into `develop` branch.

**Comment by cbowdon** [comment]
Thanks :+1:

**Pull request closed without merging**